runtime_lib/test_lib/test_library.cpp

```diff
@@ -35,15 +131,98 @@ void mlir_aie_deinit_libxaie(aie_libxaie_ctx_t *ctx) {
   if (RC != XAIE_OK) {
     printf("Failed to finish tiles.\n");
   }
+
+#ifdef HSA_RUNTIME
+  if (ctx->cmd_queue != NULL) {
+    hsa_queue_destroy(ctx->cmd_queue);
+  }
+  hsa_shut_down();
+#endif
   free(ctx);
 }
 
 /// @brief Initialize the device represented by the context.
 /// @param ctx The context
 /// @return Zero on success
-int mlir_aie_init_device(aie_libxaie_ctx_t *ctx) {
+int mlir_aie_init_device(aie_libxaie_ctx_t *ctx, uint32_t device_id) {
   AieRC RC = XAIE_OK;
 
+#ifdef HSA_RUNTIME
+  printf("Initializing device with HSA runtime\n");
```

Comment:
I'm not a fan of `printf` in library code. Especially unconditional ones the user can't turn off, but also the error `printfs` where the error is already communicated by returning -1 to the user, who can decide to `printf` or do something else.

Reply:
Thanks, addressed in [f18a8d6](https://github.com/Xilinx/mlir-aie/pull/570/commits/f18a8d66b255ed857b56380a62fe27e5ba046e25).